src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -338,68 +423,51 @@ var Marathon = React.createClass({
     }
   },
 
-  showAboutModal: function (event) {
-    if (event != null) {
-      event.preventDefault();
-    }
-
-    if (this.state.modalClass !== null) {
-      return;
-    }
-
-    this.setState({
-      modalClass: AboutModalComponent
-    });
-  },
-
-  showAppModal: function (app) {
-    if (this.state.modalClass !== null) {
-      return;
-    }
-
-    this.setState({
-      activeApp: app,
-      modalClass: AppModalComponent
-    });
-  },
+  updatePolling: function () {
+    var id = this.state.activeTabId;
 
-  showNewAppModal: function () {
-    if (this.state.modalClass !== null) {
-      return;
+    if (this.state.activeApp) {
+      this.setPollResource(this.fetchTasks);
+    } else if (id === tabs[0].id) {
+      this.setPollResource(this.fetchApps);
+    } else if (id === tabs[1].id) {
+      this.setPollResource(this.fetchDeployments);
     }
-
-    this.setState({
-      modalClass: NewAppModalComponent
-    });
   },
 
-  onTabClick: function (id) {
+  activateTab: function (id) {
     this.setState({
       activeTabId: id
     });
-
-    if (id === tabs[0].id) {
-      this.setPollResource(this.fetchApps);
-    } else if (id === tabs[1].id) {
-      this.setPollResource(this.fetchDeployments);
-    }
   },
 
-  render: function () {
-    var modal;
-
+  routeAbout: function () {
     /* jshint trailing:false, quotmark:false, newcap:false */
     /* jscs:disable disallowTrailingWhitespace, validateQuoteMarks, maximumLineLength */
-    if (this.state.modalClass === AppModalComponent) {
-      modal = (
+    return (
+      <AboutModalComponent
+        onDestroy={this.modalDestroy}
+        ref="modal" />
+    );
+    /* jshint trailing:true, quotmark:true, newcap:true */
+    /* jscs:enable disallowTrailingWhitespace, validateQuoteMarks, maximumLineLength */
+  },
+
+  routeApps: function () {
+    var activeApp = this.state.collection.get("/" + this.state.activeAppId);
+    if (activeApp) {
+      /* jshint trailing:false, quotmark:false, newcap:false */
+      /* jscs:disable disallowTrailingWhitespace, validateQuoteMarks, maximumLineLength */
+      return (
         <AppModalComponent
           activeTask={this.state.activeTask}
           appVersionsFetchState={this.state.appVersionsFetchState}
           destroyApp={this.destroyApp}
           fetchTasks={this.fetchTasks}
           fetchAppVersions={this.fetchAppVersions}
-          model={this.state.activeApp}
-          onDestroy={this.handleModalDestroy}
+          handleSetAppView={this.handleSetAppView}
```

Comment:
Yes, I am with you.
We should think about a better encapsulation, doing structural refactoring, making Marathon.jsx more leightweight, bringing function to the place where it is needed, clean up the render cycle, etc.
Let's do this in a seperate issue and discuss about it all togther.